### ISSUE TITLE
ReplicaName and Endpoints created in upper case

### ISSUE THREAD

**Issue by VickyWinner** [PowerShell/SqlServerDsc#1267]
Problem:

SqlServiceAccount does not work for SSRS 2016, it produced an internal error.

Description

We have successfully changed the service account for Database Services (DatabaseEngine) using SqlServiceAccount however i get an internal error when i try to set the service account for SSRS 2016.

Steps to produce error

SSRS 2016 EE on Windows Server 2016

$cred = Get-Credential -UserName "testdomain.local\testaccount" -Message "Password please"

Invoke-DSCResource -Module SqlServerDsc -Name SqlServiceAccount -Method Set -Property @{
serverName='TestServer''
InstanceName='TESTSSRS'
ServiceType='ReportingServices'
ServiceAccount=$cred
}

Error Produced

An internal error occurred.
+ CategoryInfo : NotSpecified: (root/Microsoft/...gurationManager:String) [], CimException
+ FullyQualifiedErrorId : HRESULT 0x8007054f
+ PSComputerName : localhost

Exploration

I have debugged the code and see that it errors out at SetServiceAccount. ( I also tried SetServiceAccount in standard powershell which also doesn't work for SSRS, but again works fine for database engine)

It looks like SetServiceAccount is not valid for reporting services and we should use SetWindowsServiceIdentity, a method of MSReportServer_ConfigurationSetting.

Has anyone successfully used SqlServiceAccount for SSRS 2016.

**Comment by VickyWinner** [comment]
Yes.  When I pass the values of hostnames for replica creation, I pass them in lower case.  But still the one replica got created in uppercase.  When I try to set AG properties using PS Provider, it couldn't find the AG replica.  So, I had to manually convert to replica name to uppercase so it can find and I can set the AG properties.  Hope this is clear now.

**Comment by johlju** [comment]
So it’s the SqlAg or SqlAGReplica that had this problem creating the replica name in upper case?
Trying to narrow down where the fault might be.
